src/modm/architecture/interface/assert.h.in

```diff
@@ -0,0 +1,87 @@
+// coding: utf-8
+/*
+ * Copyright (c) 2017, 2020, Niklas Hauser
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <modm/architecture/utils.hpp>
+
+/// @cond
+#define MODM_ASSERTION_INFO_HAS_DESCRIPTION {{ 1 if options.with_description else 0 }}
+
+// C-type information struct
+typedef struct modm_packed
+{
+	const char *name;
+%% if options.with_description
+	const char *description;
+%% endif
+	uintptr_t context;
+	uint8_t behavior;
+} _modm_assertion_info;
```

Comment:
I need to call this from C code too, hence the packed struct once in C here and once in C++ as `modm::AssertionInfo`.